### PR TITLE
fix: uSES import for react-native / esm

### LINKED ISSUE #3582: React Native: Queries not executing on v4
### Describe the bug

Hi,

I'm trying out 4.0.0-beta.7 to implement offline support in our application.

Everything is upgraded including all the breaking changes, but my queries are not being executed anymore.
To be sure it's not a problem inside my application, I've created a fresh expo app with just the App.tsx file.

When opening the app, my queries remain in a loading state. If I use the refetch of the `useQuery`,  and run this in a `useEffect`, I'm able to execute the query.

### Your minimal, reproducible example

Couldn't get react-query to work in expo snack

### Steps to reproduce

1. Create a react-native application with the following package.json file

```
{
  "name": "reactqueryv4",
  "version": "1.0.0",
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "expo start",
    "android": "expo start --android",
    "ios": "expo start --ios",
    "web": "expo start --web",
    "eject": "expo eject"
  },
  "dependencies": {
    "expo": "~44.0.0",
    "expo-status-bar": "~1.2.0",
    "react": "17.0.1",
    "react-dom": "17.0.1",
    "react-native": "0.64.3",
    "react-native-web": "0.17.1",
    "react-query": "4.0.0-beta.7"
  },
  "devDependencies": {
    "@babel/core": "^7.12.9",
    "@types/react": "~17.0.21",
    "@types/react-native": "~0.64.12",
    "typescript": "~4.3.5"
  },
  "private": true
}
```
2. Create a query with a simple async function
```
import { Text, View } from "react-native";
import { QueryClient, QueryClientProvider, useQuery } from "react-query";

export default function App() {
  const queryClient = new QueryClient();

  return (
    <QueryClientProvider client={queryClient}>
      <Todos />
    </QueryClientProvider>
  );
}

function Todos() {
  // Queries
  const query = useQuery(["todos"], async () => {
    console.log("this should log but doesn't");

    return Promise.resolve("ok");
  });

  console.log(query.data);

  return (
    <View>
      <Text>lol</Text>
    </View>
  );
}
```
3. Open your application and you will see undefined query data, a status loading and no console.log.

### Expected behavior

I expect this query to be executed. If I change my package.json to the latest v3 version, my query get's executed as it should.

### How often does this bug happen?

Every time

### Screenshots or Videos

_No response_

### Platform

- iOS, Android

### react-query version

4.0.0-beta.7

### TypeScript version

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by hirbod**
This fix is a temporary workaround to get RQ beta working with react-native again. Once uSES has fixed the bugs upstream, we can remove this workaround again.

fixes #3582

### CODE REVIEW

File: src/reactjs/useBaseQuery.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React from 'react'
-import { useSyncExternalStore } from 'use-sync-external-store/shim/index.js'
+import { useSyncExternalStore } from './useSyncExternalStore'
```
Comment (TkDodo) [inline src/reactjs/useBaseQuery.ts:2]: we have other imports from the shim as well that need to be adapted please

File: src/reactjs/useSyncExternalStore.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+// @ts-ignore
+// eslint-disable-next-line import/no-unresolved
+import { useSyncExternalStore } from 'use-sync-external-store/shim/index.js'
```
Comment (TkDodo) [inline src/reactjs/useSyncExternalStore.ts:3]: why would that import not resolve? It's the one that exists, right?

File: src/reactjs/useSyncExternalStore.native.ts [hunk 3]
```diff
@@ -0,0 +1,4 @@
+// @ts-ignore
+// eslint-disable-next-line import/no-unresolved
+import { useSyncExternalStore } from 'use-sync-external-store/shim/index.native.js'
+export { useSyncExternalStore }
```
Comment (TkDodo) [inline src/reactjs/useSyncExternalStore.native.ts:4]: could we also do:

```
export { useSyncExternalStore } from 'use-sync-external-store/shim/index.native.js'
```

Comment (hirbod) [reply src/reactjs/useSyncExternalStore.native.ts:4]: I just copied the style it was done here:
https://github.com/tannerlinsley/react-query/blob/5f80b0906b5b342782af222f75bbf78d6d267151/src/reactjs/reactBatchedUpdates.native.ts

File: src/reactjs/useBaseQuery.ts (hunk 1 above)
Comment (hirbod) [reply src/reactjs/useBaseQuery.ts:2]: done

File: src/reactjs/useSyncExternalStore.ts (hunk 2 above)
Comment (hirbod) [reply src/reactjs/useSyncExternalStore.ts]: fixed

File: src/reactjs/useSyncExternalStore.native.ts (hunk 3 above)
Comment (hirbod) [reply src/reactjs/useSyncExternalStore.native.ts]: fixed

**Comment by hirbod** [comment]
@TkDodo All change requests done

File: src/reactjs/useSyncExternalStore.ts [hunk 4]
```diff
@@ -0,0 +1,2 @@
+// @ts-ignore
+export { useSyncExternalStore } from 'use-sync-external-store/shim/index.js'
```
Comment (TkDodo) [inline src/reactjs/useSyncExternalStore.ts:2]: ts-ignore should be unneeded ?

File: src/reactjs/useSyncExternalStore.native.ts [hunk 5]
```diff
@@ -0,0 +1,2 @@
+// @ts-ignore
+export { useSyncExternalStore } from 'use-sync-external-store/shim/index.native.js'
```
Comment (TkDodo) [inline src/reactjs/useSyncExternalStore.native.ts:2]: please have a look at the prettier errors in the pipeline. I think it comes from the missing newline in both files.

File: src/reactjs/useSyncExternalStore.ts (hunk 4 above)
Comment (hirbod) [reply src/reactjs/useSyncExternalStore.ts:2]: Yeah, here it is unneeded, only needed for .native.ts

File: src/reactjs/useSyncExternalStore.native.ts (hunk 5 above)
Comment (hirbod) [reply src/reactjs/useSyncExternalStore.native.ts]: I added the newline

**Pull request merged**